core/loadOptions.js

```diff
@@ -76,7 +76,7 @@ var legacyOptionsChecker = function(options, fileName){
 module.exports = function(basePath, _silent){
     silent = _silent;
 
-    var pathToApp = basePath || path.dirname(require.main.filename);
+    var pathToApp = basePath || global.pathToApp;
```

Comment:
Globals are indeed a bad practice.
However, beyond my reasoning for running from other location, it was also inconsistent with the rest of the SourceJS code base.

I used the **global.pathToApp**, since it was used by the rest of the **core** files (see processMd, specUtils, etc). =)